File: android/src/main/kotlin/com/sendsay/data/SSECEvent.kt
package com.sendsay.data

import com.sendsay.sdk.models.TrackSSECData
import com.sendsay.sdk.models.TrackingSSECType

data class SSECEvent(val type: TrackingSSECType, val data: TrackSSECData) {
    companion object {
        fun fromMap(map: Map<String, Any?>): SSECEvent {
            return SSECEvent(
                type = map["type"] as TrackingSSECType? ?: throw IllegalStateException("SSECEvent.type is required!"),
                data = map["data"] as TrackSSECData? ?: throw IllegalStateException("SSECEvent.data is required!")
            )
        }
    }
}

File: lib/src/data/model/ssec_event.dart
import 'package:meta/meta.dart';
import 'package:sendsay/sendsay.dart';

@immutable
class SSECEvent {
  final TrackingSSECType type;
  final TrackSSECData data;

  const SSECEvent({
    required this.type,
    required this.data,
  });

  @override
  String toString() {
    return 'SSECEvent{'
        'type: $type, '
        'data: $data'
        '}';
  }
}
